### PR TITLE
Update privacy toggle to work on a per site basis

### PR DESCRIPTION
**Pull request by subsymbolic**
Task/Issue URL: https://app.asana.com/0/72649045549333/559623300949084
Tech Design URL: N/A
CC:

**Description**:
Update privacy toggle to work on a per site basis.

**Steps to test this PR**:

**TEST 1: Test new toggle text and behavior**
1. Navigate to a page e.g cnn.com
1. Open the privacy dashboard and ensure that the toggle text is "Site Privacy Protection"
1. Switch privacy protection off and press _Done_
1. Ensure page reloads
1. Ensure page is now in whitelist
1. Ensure that mp_wla pixel is fired
1. Navigate to another page e.g facebook.com
1. Check that privacy toggle is on
1. Navigate back cnn.com and ensure that the privacy/site protection toggle is back off
1. Switch the toggle back on
1. Ensure that  mp_wlr pixel is fired
1. Ensure page is no longer in whitelist

**TEST 2: Test new buttons**
1. Navigate to a page and open the privacy Dashboard
1. Tap the _Manage Whitelist_ button and ensure whitelist page is launched
1. Ensure the "mp_mw" pixel fires
1. Go back
1. Tap the _Report Broken Site_ button and ensure that the broken site button in launched
1. Ensure the "mp_rb" pixel fires

**TEST 3: Test new pixels browsing pixels**
1. Browse to a site
1. From the browsing menu, add it to the whitelist and ensure that "mb_wla" fires
1. From the browsing menu, remove it from the whitelist and ensure that "mb_wlr" fires

**TEST 4: Test new settings pixel**
1. Open _Settings_
1. Tap _Privacy Protection Whitelist_
1. Ensure "ms_mw" fires

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/PrivacyProtectionFooterController.swift [hunk 1]
```diff
@@ -22,28 +22,42 @@ import UIKit
...
         super.viewDidLoad()
         leaderboard.didLoad()
+        update(forHeight: view?.frame.size.height ?? 0)
     }
-
-    override func viewWillAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        update()
+    
+    func update(forHeight height: CGFloat) {
```
Comment (brindy) [inline DuckDuckGo/PrivacyProtectionFooterController.swift]: I'm not clear why this change was made?  The screen now seems to be badly laid out. 

Simulator Screen Shot - iPhone 11 Pro - 2019-10-28 at 11 02 44

Simulator Screen Shot - iPhone 11 Pro - 2019-10-28 at 11 01 32

**Comment by bwaresiak** [comment]
Some UX comments from me:
1. Opening Privacy Protection screen for page that is whitelisted result in late switch animation.
2. Since we have this, could we remove Add to/Remove from Whitelist buttons from the Menu?

**Review by brindy — CHANGES_REQUESTED** [review]
1) The leaderboard is not centred on the iPhone 11 pro, by quite a bit.  It is also not perfect on the iPhone 5s, but passable.  See images below.

2) Launching the dashboard on the iPad crashes (test on simulator iPad Pro 12.9, iOS 13.1)

3) The extensions show a prompt when the user disables site protection. "Is this site broken? Yes | No " which then launches the broken site report if the user clicks yes.  Do we need to do that here too?

**Images:**

iPhone 11 Pro - landscape

iPhone 11 Pro - portrait

**Comment by subsymbolic** [comment]
@bwaresiak thanks for your feedback!
1. Opening Privacy Protection screen for page that is whitelisted result in late switch animation: This is preexisting toggle behavior and so beyond the scope of this PR which involves minimal visual changes. I _did_ have a quick look as I thought the animation just needed to be turned off however that sadly didn't do the job. I'll log it as a low hanging fruit task
2. Since we have this, could we remove Add to/Remove from Whitelist buttons from the Menu? That is a very good idea! I have added pixels to those items to see how much they are used versus the toggle. We can make our decision when those pixels are in. I have added a task to the project to follow up on this after release.

**Comment by subsymbolic** [comment]
@brindy that's good feedback. The screen had some margin on it (rather than being centred). I have returned to that approach (which fixes the tablet crash too), although with a smaller margin as there is more on the screen now.

File: Core/Pixel.swift [hunk 2]
```diff
@@ -66,6 +68,7 @@ public enum PixelName: String {
     case settingsHomePageSimple = "ms_hp_s"
     case settingsHomePageCenterSearch = "ms_hp_c"
     case settingsHomePageCenterSearchAndFavorites = "ms_hp_f"
+    case settingsdManageWhitelist = "ms_mw"
```
Comment (brindy) [inline Core/Pixel.swift]: typo: `settingsdManageWhitelist` -> `settingsManageWhitelist`

**Review by brindy — COMMENTED** [review]
Looking really good now!  Just one typo in the pixels, but on the dashboard when viewing on iPad if the user goes to the whitelist there's no "back" like there are on other screens.

**Review by brindy — APPROVED** [review]
Good job! LGTM!

**Pull request merged**